### PR TITLE
[MBL-15412][All] Target Android 11

### PR DESCRIPTION
**Pull request by tamaskozmer**
Android 11 behavioural changes: https://developer.android.com/about/versions/11/privacy
What affects us from these is the package visibility and storage changes.

Test plan:
- Smoke test all the apps

- Package visibility: Test if the student view can be opened from the teacher app

- Storage changes: Check the most common storage use cases on both Android 10 and Android 11

Storage scenarios:

- Downloading/uploading files from/to the Canvas files
- Adding and downloading attachments in messages
- Submitting a file to an assignment (student only)
- Submitting a file to an assignment comment
- Embedding an image from the device to an RCE content (for example text submission in student, edit page in teacher)
- Adding profile picture from a file on the device (student only)
- Creating a panda avatar and saving it on the device (student only)
- Adding a file to Studio from the device

refs: MBL-15412
affects: All
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

File: libs/login-api-2/src/main/java/com/instructure/loginapi/login/activities/BaseLoginSignInActivity.kt [hunk 1]
```diff
@@ -100,7 +100,7 @@ abstract class BaseLoginSignInActivity : AppCompatActivity(), OnAuthenticationSe
     private var httpAuthHandler: HttpAuthHandler? = null
     private var shouldShowProgressBar = false
 
-    private val accountDomain: AccountDomain by lazy { intent.getParcelableExtra<AccountDomain>(ACCOUNT_DOMAIN) }
+    private val accountDomain: AccountDomain by lazy { intent.getParcelableExtra<AccountDomain>(ACCOUNT_DOMAIN)!! }
```
Comment (hermannakos) [inline libs/login-api-2/src/main/java/com/instructure/loginapi/login/activities/BaseLoginSignInActivity.kt:103]: Is it sure it's not going to be null?

File: libs/pandautils/src/main/java/com/instructure/pandautils/services/NotoriousUploadService.kt [hunk 2]
```diff
@@ -104,7 +104,7 @@ class NotoriousUploadService : IntentService(NotoriousUploadService::class.java.
 
         createNotificationChannel(CHANNEL_ID)
 
-        mediaPath = intent.getStringExtra(Const.MEDIA_FILE_PATH)
+        mediaPath = intent.getStringExtra(Const.MEDIA_FILE_PATH) ?: ""
```
Comment (hermannakos) [inline libs/pandautils/src/main/java/com/instructure/pandautils/services/NotoriousUploadService.kt:107]: Should we handle if it's null?

**Comment by inst-danger** [comment]
<p>Parent Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Review by hermannakos — APPROVED** [review]

**Pull request merged**